### ISSUE TITLE
[vcpkg/scripts] move toolchain into triplet?

### ISSUE THREAD

**Issue by Neumann-A** [microsoft/vcpkg#11546]
The question is if we should move:
https://github.com/microsoft/vcpkg/blob/2ac2b5f3da6b46ff2ae900550bd7c44f39c57dcd/scripts/cmake/vcpkg_configure_cmake.cmake#L196-L214

into the triplet instead an just define `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` there. (would also require to set `VCPKG_LOAD_VCVARS_ENV` there so that VCPKG loads the environment for windows)
This also applies to the other configure scripts. This would make the use of the used toolchains more transparent for the outside and shows how to use a custom toolchain better. 
We then should probably also move toolchains out of scripts into the vcpkg_root or maybe even into triplets?

**Comment by BillyONeal** [comment]
Taking off my maintainer hat and speaking personally for a moment: I think being explicit about this in the triplet file would be an improvement because it removes a 'spooky action at a distance' point where the tool is figuring this out for you behind the scenes.

On the other hand this is likely to cause users to edit the toolchain scripts which are not intended as an extension / modification point, and we wouldn't be able to support user modifications like that.

Putting my maintainer hat back on again: While I would personally love a change like this I don't think it overcomes the usual "every feature starts out with -1000 points" test, so I don't think we would want to go in this direction.

That said, we should certainly improve error messages and documentation, for example 

https://github.com/microsoft/vcpkg/blob/85247c398141ab4d692b8fd61dbfcb845e79d2f2/toolsrc/src/vcpkg/build.cpp#L640

to point users at `VCPKG_CHAINLOAD_TOOLCHAIN_FILE`.

**Comment by strega-nil-ms** [comment]
I don't think it would be possible to do this as of right now; we could deprecate and do this on our triplets, but users may be relying on this behavior in their own custom triplets.
